pymodbus/pdu/bit_message.py

```diff
@@ -23,7 +23,7 @@ def encode(self) -> bytes:
 
     def decode(self, data: bytes) -> None:
         """Decode a request pdu."""
-        self.address, self.count = struct.unpack(">HH", data)
```

Comment:
According to the modbus standard, a read_coil request contains address and count, nothing more.

pymodbus support the standard, not non-standard messages !

if you receive a non-standard message (which happens in real life), then please modify it in the trace_packet hook and pass a legal packet.